src/app/modules/group/components/group-remove-button/group-remove-button.component.ts

```diff
@@ -1,26 +1,30 @@
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, Input, OnChanges } from '@angular/core';
 import { Group } from '../../http-services/get-group-by-id.service';
-import { catchError, map, tap } from 'rxjs/operators';
-import { BehaviorSubject, merge, Observable, of } from 'rxjs';
+import { distinct, switchMap, map } from 'rxjs/operators';
+import { Observable, ReplaySubject, Subject } from 'rxjs';
 import { GetGroupChildrenService, GroupChild } from '../../http-services/get-group-children.service';
 import { ConfirmationService } from 'primeng/api';
 import { ActionFeedbackService } from '../../../../shared/services/action-feedback.service';
 import { GroupDeleteService } from '../../services/group-delete.service';
 import { Router } from '@angular/router';
-
-type GroupChildrenState = 'loading' | 'hasChildren' | 'empty' | 'error';
+import { mapToFetchState } from '../../../../shared/operators/state';
 
 @Component({
   selector: 'alg-group-remove-button',
   templateUrl: './group-remove-button.component.html',
   styleUrls: [ './group-remove-button.component.scss' ]
 })
-export class GroupRemoveButtonComponent implements OnInit {
+export class GroupRemoveButtonComponent implements OnChanges {
   @Input() group!: Group;
```

Comment:
In this project, we never use force unwrap for inputs as you cannot guarantee the group parameter is really given.